pallets/circuit/src/state/mod.rs

```diff
@@ -60,38 +61,42 @@ pub enum InsuranceEnact {
 }
 
 impl CircuitStatus {
-    fn determine_insurance_status<T: Config>(
-        side_effect_id: SideEffectId<T>,
-        insurance_deposits: &[(
-            SideEffectId<T>,
-            InsuranceDeposit<T::AccountId, T::BlockNumber, EscrowedBalanceOf<T, T::Escrowed>>,
-        )],
+    fn determine_fsx_bidding_status<T: Config>(
+        fsx: FullSideEffect<T::AccountId, T::BlockNumber, EscrowedBalanceOf<T, T::Escrowed>>,
     ) -> CircuitStatus {
-        return if let Some((_id, insurance_request)) = insurance_deposits
-            .iter()
-            .find(|(id, _)| *id == side_effect_id)
-        {
-            if insurance_request.bonded_relayer.is_some() {
-                CircuitStatus::Bonded
-            } else {
-                CircuitStatus::PendingInsurance
+        if let Some(bid) = fsx.best_bid {
+            if fsx.security_lvl == SecurityLvl::Optimistic
+                && (bid.optimistic_insurance.is_none() || bid.reserved_bond.is_none())
+            {
+                // todo: handle
+                panic!();
+            } else if fsx.security_lvl == SecurityLvl::Escrowed
+                && (bid.optimistic_insurance.is_some() || bid.reserved_bond.is_some())
```

Comment:
we still require an insurance deposit for escrow. Otherwise, we have no way of fining/slashing optimistic executors if they bid on an execution but then don't execute them.